src/main/java/com/example/libraryapi/service/ScheduledTaskService.java

```diff
@@ -0,0 +1,48 @@
+package com.example.libraryapi.service;
+
+import com.example.libraryapi.model.Reservation;
+import com.example.libraryapi.repository.ReservationItemRepository;
+import com.example.libraryapi.repository.ReservationRepository;
+import jakarta.transaction.Transactional;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Caching;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ScheduledTaskService {
+
+    private final ReservationRepository reservationRepository;
+    private final ReservationItemRepository reservationItemRepository;
+
+    @Caching(
+            evict = {
+                    @CacheEvict(value = "books_cache", allEntries = true),
+                    @CacheEvict(value = "book_cache", allEntries = true),
+                    @CacheEvict(value = "books_by_category_cache", allEntries = true),
+                    @CacheEvict(value = "books_by_phrase_cache", allEntries = true),
+                    @CacheEvict(value = "reservation_items_cache", allEntries = true),
+                    @CacheEvict(value = "reservation_item_by_client_cache", allEntries = true),
+            }
+    )
+    @Scheduled(cron = "${com.scheduled.cron}")
+    public void deleteExpiredReservations() {
+        LocalDate today = LocalDate.now();
+        System.out.println("Today: " + today);
```

Comment:
Dodałbym logowanie tutaj, możesz użyć loggera z https://projectlombok.org/features/log

Reply:
@Franek-Antoniak czy to jest coś podobnego do ustawienia logów w propertiesach na repo/serwis?

Reply:
@marysiakodziara a co to o czym mówisz, to logowanie z lomboka jest mega przyjemne dajesz @Slf4j na klasę i potem tam gddzie chce coś wypisać to dajesz log.debug("hellom") a jakby był error i nie chcesz rzucać wyjątku ale chcesz żeby zostało w logach to: log.error("Wtf")